pkg/utils/dataset/lifecycle/node.go

```diff
@@ -152,29 +152,32 @@ func LabelCacheNode(nodeToLabel v1.Node, runtimeInfo base.RuntimeInfoInterface,
 			toUpdate.Labels = make(map[string]string)
 		}
 
-		toUpdate.Labels[runtimeLabel] = "true"
-		updatedLabels = append(updatedLabels, runtimeLabel)
-		toUpdate.Labels[commonLabel] = "true"
-		updatedLabels = append(updatedLabels, commonLabel)
+		labelsToModify.Add(runtimeLabel, "true", common.AddLabel)
+		labelsToModify.Add(commonLabel, "true", common.AddLabel)
+
 		if exclusiveness {
-			toUpdate.Labels[exclusiveLabel] = utils.GetExclusiveValue(runtimeInfo.GetNamespace(), runtimeInfo.GetName())
-			updatedLabels = append(updatedLabels, exclusiveLabel)
+			exclusiveLabelValue := utils.GetExclusiveValue(runtimeInfo.GetNamespace(), runtimeInfo.GetName())
+			labelsToModify.Add(exclusiveLabel, exclusiveLabelValue, common.AddLabel)
 		}
 
 		if currentDatasetNum, ok := toUpdate.Labels[datasetLabel]; ok {
 			currentData, err := strconv.Atoi(currentDatasetNum)
 			if err != nil {
 				return err
 			}
-			toUpdate.Labels[datasetLabel] = strconv.Itoa(currentData + 1)
+			datasetLabelValue := strconv.Itoa(currentData + 1)
```

Comment:
Since we got a `DecreaseDatasetNum` func, maybe we should have another `IncreaseDatasetNum` func here?

Reply:
Fixed it. Thanks.

Reply:
FIxed it. Thanks.